apps/dashboard/test/models/manifest_test.rb

```diff
@@ -126,4 +126,43 @@ class ManifestTest < ActiveSupport::TestCase
     assert_equal "", manifest.url
     assert_equal({}, manifest.metadata)
   end
+
+  test 'manifest only writes valid entries' do
+    opts = { name: 'Ruby', 
+             description: 'test hash', 
+             icon: 'fas://cog',
+             url: '/some/location',
+             category: 'some category',
+             subcategory: 'some subcategory',
+             role: 'some role',
+             metadata: { some_key: 'value' },
+             new_window: false,
+             caption: 'some caption',
+             invalid_key: 'invalid value'
+          }
+
+    valid_manifest = <<~HEREDOC
+      ---
+      name: Ruby
+      description: test hash
+      icon: fas://cog
+      url: "/some/location"
+      category: some category
+      subcategory: some subcategory
+      role: some role
+      metadata:
+        some_key: value
+      new_window: false
+      caption: some caption
+    HEREDOC
+
+    manifest = Manifest.new(opts)
+
+    Dir.mktmpdir do |dir|
+      path = dir << '/manifest.yml'
+      manifest.save(path)
+
+      assert_equal valid_manifest, File.read(path)
+    end
```

Comment:
Does this work too? (needs some imports probably?). I looked up where this was tripping us up and it's this object type that's being passed. duck type testing may be fine, but I don't know, it's cheap to add the check,

```suggestion
      assert_equal valid_manifest, File.read(path)
      
      manifest = Manifest.new(ActionController::Paramaters.new(param: "shouldn't write me").merge(opts))
      manifest.save(path)
      assert_equal valid_manifest, File.read(path)
    end
```

Reply:
It throws an error before ever getting to the `save` in the test. The error that comes up is `ActionController::UnfilteredParameters: unable to convert unpermitted parameters to hash` which seems to be coming from enforcing strong parameters.

But I'm not sure where those strong parameters are defined which it would like. Even passing just a hash that has all the expected fields causes  the error to come up.

Reply:
it's fine if it ends up being too hard to test.